### PR TITLE
Adds a method to get all ids for all resources in the table

### PR DESCRIPTION
**Pull request by shubham1g5**
cross-request: https://github.com/dimagi/commcare-android/pull/2335

### CODE REVIEW

File: src/main/java/org/commcare/resources/model/ResourceTable.java [hunk 1]
```diff
@@ -1208,4 +1209,8 @@ public void setMissingResources(SizeBoundUniqueVector<Resource> missingResources
     public Vector<Integer> getLazyResourceIds() {
         return storage.getIDsForValue(Resource.META_INDEX_LAZY, LAZY_VAL_TRUE);
     }
+
+    public List<Integer> getResourceIds() {
```
Comment (ShivamPokhriyal) [inline src/main/java/org/commcare/resources/model/ResourceTable.java:1213]: nit: How about `getAllReourceIds`?

**Review by ShivamPokhriyal — DISMISSED** [review]

File: src/main/java/org/javarosa/core/services/storage/IStorageUtilityIndexed.java [hunk 2]
```diff
@@ -150,6 +150,12 @@
      */
     Object getAccessLock();
 
+
+    /**
+     * @return a Vector of IDs of all Externalizable objects in storage
+     */
+    Vector<Integer> getAllIDs();
```
Comment (ShivamPokhriyal) [inline src/main/java/org/javarosa/core/services/storage/IStorageUtilityIndexed.java:157]: Wondering whether it needs to be implemented on formplayer?

**Review by ctsims — CHANGES_REQUESTED** [review]
I don't think that this change should warrant changing the IStorageUtilityIndexed interface / contract, which we should consider a breaking change for implementing platforms.

The getAllIDs() method here could be replicated by either doing an iterate with the 'false' flag set (to ensure that records don't get sent down) and pulling out the integers (which is what the implementation ends up doing) or by calling getIDsForValues() with empty String inputs as filters (which is how this would work in a sql-ish pattern). My guess is that the latter is the safest / best implementation for our needs, and that if it the current code doesn't run that way it's safe to change it to function that way, based on the method contract.

**Comment by shubham1g5** [comment]
@ctsims Updated based on the suggestion.

**Review by ctsims — APPROVED** [review]

File: src/main/java/org/commcare/modern/database/DatabaseHelper.java [hunk 3]
```diff
@@ -34,6 +34,11 @@
     public static Pair<String, String[]> createWhere(String[] fieldNames, Object[] values,
```
Comment (ctsims) [inline src/main/java/org/commcare/modern/database/DatabaseHelper.java:34]: Can we add a test to confirm that the underlying endpoints that use this return all results and not an error, so the behavior can be regressed?

Comment (shubham1g5) [reply src/main/java/org/commcare/modern/database/DatabaseHelper.java:34]: Good point, does [this](https://github.com/dimagi/commcare-core/pull/930/commits/7cb5743d401602d9382cc2e8b8b242878b8b4985) look good ?

File: src/test/java/org/javarosa/core/storage/IndexedStorageUtilityTests.java [hunk 4]
```diff
@@ -131,22 +128,32 @@ public void testBulkMetaMatching() {
...
     }
 
+    @Test
+    public void testReadingAllEntries() {
+        writeBulkSets();
+        List<Integer> matches = storage.getIDsForValues(new String[]{}, new String[]{});
+        Set<Integer> expectedMatches = getIdsFromModels(tenSizesOfMensNikes);
+        expectedMatches.addAll(getIdsFromModels(eightSizesOfWomensNikes));
+        expectedMatches.addAll(getIdsFromModels(fiveSizesOfMensVans));
+        Assert.assertEquals("Failed index match [brand,style][nike,mens]", expectedMatches, new HashSet<>(matches));
```
Comment (ctsims) [inline src/test/java/org/javarosa/core/storage/IndexedStorageUtilityTests.java:138]: I think this assert has the wrong text.

File: src/test/java/org/javarosa/core/storage/IndexedStorageUtilityTests.java [hunk 5]
```diff
@@ -131,22 +128,32 @@ public void testBulkMetaMatching() {
         Assert.assertEquals("Failed index match [brand,style][nike,mens]", getIdsFromModels(tenSizesOfMensNikes), getIdsFromModels(matchedRecords.toArray(new Shoe[]{})));
     }
 
+    @Test
+    public void testReadingAllEntries() {
+        writeBulkSets();
+        List<Integer> matches = storage.getIDsForValues(new String[]{}, new String[]{});
```
Comment (ctsims) [inline src/test/java/org/javarosa/core/storage/IndexedStorageUtilityTests.java:134]: Style nit: You can also instantiate this with `new String[0]` which might make the intent slightly clearer

**Review by ctsims — DISMISSED** [review]
Left a couple tiny point of feedback, but the new test looks good (And thanks for the linting)

**Review by ctsims — APPROVED** [review]

**Pull request merged**